### PR TITLE
reponsive dashboard page

### PR DESCRIPTION
**Pull request by nhathoang241199**
- Responsive dashboard page.

### CODE REVIEW

File: src/pages/Dashboard/components/EventInfo.tsx [hunk 1]
```diff
@@ -87,35 +90,47 @@ const EventInfo = () => {
...
+                mr={{ base: 2, tablet: 4 }}
+              >
                 <Image
                   src="/assets/layout/notification.svg"
                   alt="notification"
                 />
-              </Box>
-              <VStack flex={8} alignItems="flex-start">
+              </Center>
+              <Box sx={eventContent} alignItems="flex-start">
```
Comment (lhtrung307) [inline src/pages/Dashboard/components/EventInfo.tsx:119]: `alignItems="flex-start"` without `flex`?

File: src/pages/Dashboard/components/EventInfo.tsx [hunk 2]
```diff
@@ -70,8 +73,8 @@ const EventInfo = () => {
   }, [api?.query.system]);
 
   return (
-    <Card flex={4}>
-      <VStack alignItems="flex-start">
+    <Card display="flex" flexDirection="column">
+      <Box alignItems="flex-start">
```
Comment (lhtrung307) [inline src/pages/Dashboard/components/EventInfo.tsx:77]: `flex-start` without `flex`?

File: src/pages/Dashboard/components/EventInfo.tsx [hunk 3]
```diff
@@ -70,8 +73,8 @@ const EventInfo = () => {
   }, [api?.query.system]);
 
   return (
-    <Card flex={4}>
-      <VStack alignItems="flex-start">
+    <Card display="flex" flexDirection="column">
```
Comment (lhtrung307) [inline src/pages/Dashboard/components/EventInfo.tsx:76]: `Card` already has this style.

File: src/pages/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -71,51 +72,58 @@ const Dashboard = () => {
...
+          <Card h="full">
+            <VStack alignItems="flex-start">
+              <Text mb={4} fontWeight="bold" color="primary">
+                {nodeInfo?.nodeName}
+              </Text>
+              <Text fontSize="sm" fontWeight="light">
+                {nodeInfo?.chain}
+              </Text>
+              <Text color="black">{socket}</Text>
+              <HStack w="full" py={5} borderTop="1px solid #EEF1FF">
```
Comment (lhtrung307) [inline src/pages/Dashboard/Dashboard.tsx:97]: Color variable.

**Review by lhtrung307 — CHANGES_REQUESTED** [review]

File: src/components/SkeletonLoadingRow/SkeletonLoadingRow.tsx [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 import React from 'react';
 
 interface IProps {
   columnAmount: number;
 }
 
 const SkeletonLoadingRow: React.FC<IProps> = ({ columnAmount }) => {
-  const columnArray = new Array(columnAmount).fill(0);
+  const [isMobile] = useMediaQuery('(max-width: 739px)');
+  const columnArray = new Array(isMobile ? 3 : columnAmount).fill(0);
```
Comment (lhtrung307) [inline src/components/SkeletonLoadingRow/SkeletonLoadingRow.tsx:10]: Should move this logic outside of component.

File: src/components/pagination/Pagination.tsx [hunk 6]
```diff
@@ -60,16 +68,40 @@ const Pagination = (props: IProps) => {
...
       : currentPage * resultsPerPage;
 
   return (
-    <Flex justifyContent="space-between" alignItems="center">
-      <Flex pl={8} flex="4" justifyContent="flex-start">
+    <Flex
+      justifyContent="space-between"
+      flexDirection={{
+        base: 'column',
+        tablet: isSmallScreen ? 'column' : 'row',
```
Comment (lhtrung307) [inline src/components/pagination/Pagination.tsx:85]: Using double media query? `tablet` is a media query already.

File: src/components/pagination/Pagination.tsx [hunk 7]
```diff
@@ -94,7 +126,7 @@ const Pagination = (props: IProps) => {
       </Flex>
       <Flex flex="14" justifyContent="flex-end">
         <Skeleton isLoaded={!isLoading}>
-          {!!totalCount && (
+          {!!totalCount && isMobile ? null : (
```
Comment (lhtrung307) [inline src/components/pagination/Pagination.tsx:129]: Create a variable for this.

File: src/components/sideBar/Faucet.tsx [hunk 8]
```diff
@@ -26,6 +27,7 @@ const Faucet = () => {
       setIsLoading(false);
     } else {
       toast({
+        position: 'top-right',
```
Comment (lhtrung307) [inline src/components/sideBar/Faucet.tsx:30]: Should consistence with another toast.

File: src/contexts/substrateContext/substrateContext.tsx [hunk 9]
```diff
@@ -278,6 +278,7 @@ const useSubstrate = () => {
   const { state, dispatch } = useContext(SubstrateContext);
   const toast = useToast();
   const { t } = useTranslation();
+  const [isMobile] = useMediaQuery('(max-width: 739px)');
```
Comment (lhtrung307) [inline src/contexts/substrateContext/substrateContext.tsx:281]: Too many type of media query. Should follow defined breakpoints.

File: src/hooks/useChangeOwner.ts [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import { useToast } from '@chakra-ui/react';
+import { useMediaQuery, useToast } from '@chakra-ui/react';
```
Comment (lhtrung307) [inline src/hooks/useChangeOwner.ts:1]: `useMediaQuery` is not used anywhere. Remove it.

File: src/hooks/useClaimContract.ts [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import { useToast } from '@chakra-ui/react';
+import { useMediaQuery, useToast } from '@chakra-ui/react';
```
Comment (lhtrung307) [inline src/hooks/useClaimContract.ts:1]: Same here.

File: src/hooks/useCreatePool.ts [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import { useToast } from '@chakra-ui/react';
+import { useMediaQuery, useToast } from '@chakra-ui/react';
```
Comment (lhtrung307) [inline src/hooks/useCreatePool.ts:1]: Same here.

File: src/hooks/useFaucet.ts [hunk 13]
```diff
@@ -13,11 +13,13 @@ const useFaucet = () => {
   } = useSubstrate();
   const toast = useToast();
   const { t } = useTranslation();
+  const [isMobile] = useMediaQuery('(max-width: 739px)');
```
Comment (lhtrung307) [inline src/hooks/useFaucet.ts:16]: Do we need this?

File: src/hooks/usePool.ts [hunk 14]
```diff
@@ -94,8 +105,12 @@ const useUpfrontPool = () => {
     }
   };
 
-  const leavePool = async (poolPackage: string) => {
-    setLoadingPool(poolPackage);
+  const leavePool = async (poolPackage?: string) => {
+    if (poolPackage) {
+      setLoadingPool(poolPackage || '');
```
Comment (lhtrung307) [inline src/hooks/usePool.ts:110]: Why do we have to use `|| ''` here?

File: src/hooks/usePool.ts [hunk 15]
```diff
@@ -129,12 +146,41 @@ const useUpfrontPool = () => {
     }
   };
 
+  const mutation = useMutation(
```
Comment (lhtrung307) [inline src/hooks/usePool.ts:149]: Too many things in one hook.

File: src/pages/Dashboard/components/EventInfo.tsx [hunk 16]
```diff
@@ -73,64 +77,58 @@ const EventInfo = () => {
...
+            <Icon w={18} h={18}>
+              <path fill="currentColor" d={mdiTrashCanOutline} />
+            </Icon>
+          }
+        />
+      </HStack>
+      {React.Children.toArray(
+        eventFeed?.map((event, index: number) => (
+          <HStack
+            borderTop={
```
Comment (lhtrung307) [inline src/pages/Dashboard/components/EventInfo.tsx:106]: Should check the stack divider feature.

File: src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx [hunk 17]
```diff
@@ -1,49 +1,59 @@
...
+  Link,
   Td,
   Text,
   Tr,
   useColorModeValue,
+  useMediaQuery,
   VStack,
 } from '@chakra-ui/react';
 import { formatBalance } from '@polkadot/util';
+import { t } from 'i18next';
```
Comment (lhtrung307) [inline src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx:13]: Wrong import.

File: src/pages/SponsoredPool/SponsoredPool.tsx [hunk 18]
```diff
@@ -1,181 +1,114 @@
...
+            ) : undefined
+          }
+          onClick={onOpen}
+        >
+          {t('ADD_POOL')}
+        </Button>
+      </HStack>
+      <SponsoredPoolTable
+        captions={
+          isMobile || isSmallScreen
```
Comment (lhtrung307) [inline src/pages/SponsoredPool/SponsoredPool.tsx:79]: Maybe we can have the `display` attribute for the table column.

File: src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx [hunk 19]
```diff
@@ -1,49 +1,59 @@
...
       <Tr cursor="pointer" onClick={onClick}>
-        <Td textAlign="center" minWidth={{ sm: '250px' }}>
+        <Td textAlign="center">
           <HStack>
             <Avatar
-              mr={4}
-              w={14}
-              h={14}
+              mr={{ base: 0, pc: 4 }}
+              w={{ base: 10, pc: isSmallScreen ? 10 : 14 }}
```
Comment (lhtrung307) [inline src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx:55]: Double media query.

File: src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx [hunk 20]
```diff
@@ -54,39 +64,60 @@ const SponsoredPoolTableRow: React.FC<IProps> = ({
...
-        </Td>
-        <Td textAlign="center" fontWeight="normal" fontSize="md">
-          {isOwned ? (
-            <OwnedTableActions poolId={pool.id} onClick={onEditClick} />
-          ) : (
-            <TableActions poolId={pool.id} />
-          )}
-        </Td>
+        {isMobile || isNormalScreen ? (
+          <Td
```
Comment (lhtrung307) [inline src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx:87]: We can use detail as table action.

File: src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx [hunk 21]
```diff
@@ -54,39 +64,60 @@ const SponsoredPoolTableRow: React.FC<IProps> = ({
...
+              <Text fontWeight="normal" fontSize="md" color={textColor}>
+                {txLimit}
+              </Text>
+            </Td>
+            <Td textAlign="center" maxWidth="130px">
+              <Text fontWeight="normal" fontSize="md" color={textColor}>
+                {formatBalance(
+                  amount,
+                  { withSi: true, forceUnit: '-', withUnit: '' },
+                  chainDecimal || 18
```
Comment (lhtrung307) [inline src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx:108]: `chainDecimal` already have the default value.

File: src/utils/utils.ts [hunk 22]
```diff
@@ -119,8 +119,9 @@ export const getFromAcct = async (
   return [address, { signer: injector.signer }];
 };
 
-export const shorten = (hash: string) => {
+export const shorten = (hash: string, length?: number) => {
   const n = hash.length;
+  if (length) return `${hash.substr(0, length)}...${hash.substr(n - length)}`;
```
Comment (lhtrung307) [inline src/utils/utils.ts:124]: We can provide the default value for length as 6

**Review by lhtrung307 — CHANGES_REQUESTED** [review]

File: src/components/SkeletonLoadingRow/SkeletonLoadingRow.tsx (hunk 5 above)
Comment (nhathoang241199) [reply src/components/SkeletonLoadingRow/SkeletonLoadingRow.tsx:10]: Can you suggest a place to move

Comment (lhtrung307) [reply src/components/SkeletonLoadingRow/SkeletonLoadingRow.tsx:10]: You can move it to the parent component.

File: src/components/pagination/Pagination.tsx (hunk 6 above)
Comment (nhathoang241199) [reply src/components/pagination/Pagination.tsx:85]: I want in small screen [1024px - 1400px] , the direction is 'column'. Tablet is (>= 740)

File: src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx (hunk 19 above)
Comment (nhathoang241199) [reply src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx:55]: pc (> 1024px) isn't same pc and smallScreen (> 1024px && (1024 - 1400))

File: src/components/pagination/Pagination.tsx (hunk 6 above)
Comment (lhtrung307) [reply src/components/pagination/Pagination.tsx:85]: If you want that, you can write like this: `tablet: 'row', pc: 'column'`.

File: src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx (hunk 19 above)
Comment (lhtrung307) [reply src/pages/SponsoredPool/components/SponsoredPoolTableRow.tsx:55]: Same here. It's just the way you use it.

File: src/components/pagination/Pagination.tsx (hunk 6 above)
Comment (nhathoang241199) [reply src/components/pagination/Pagination.tsx:84]: but pc is > 1024px. I don't want the screen larger than 1400px is styled.

**Review by lhtrung307 — APPROVED** [review]

**Pull request merged**